blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/serializers/Serializers.java

```diff
@@ -0,0 +1,254 @@
+    public static SetRollupSerializer setRollupInstance = new SetRollupSerializer();
+    public static GaugeRollupSerializer gaugeRollupInstance = new GaugeRollupSerializer();
+    public static CounterRollupSerializer counterRollupInstance = new CounterRollupSerializer();
+    public static EnumRollupSerializer enumRollupInstance = new EnumRollupSerializer();
+
+    static class Type {
+        static final byte B_ROLLUP = (byte)'r';
+        static final byte B_FLOAT_AS_DOUBLE = (byte)'f';
+        static final byte B_ROLLUP_STAT = (byte)'t';
+        static final byte B_COUNTER = (byte)'C';
+        static final byte B_TIMER = (byte)'T';
+        static final byte B_SET = (byte)'S';
+        static final byte B_GAUGE = (byte)'G';
+        static final byte B_ENUM = (byte)'E';
+    }
+    
+    /** return a serializer for a specific type */
+    public static <T> AbstractSerializer<T> serializerFor(Class<T> type) {
+        if (type == null)
+            throw new RuntimeException("serializable type cannot be null",
+                    new SerializationException("serializable type cannot be null"));
+        else if (type.equals(String.class))
+            throw new RuntimeException("We don't serialize strings anymore",
+                    new SerializationException("We don't serialize strings anymore"));
+        
+        if (type.equals(BasicRollup.class))
+            return (AbstractSerializer<T>) basicRollupInstance;
+        else if (type.equals(BluefloodTimerRollup.class))
+            return (AbstractSerializer<T>)timerRollupInstance;
+        else if (type.equals(HistogramRollup.class))
+            return (AbstractSerializer<T>) HistogramSerializer.get();
+        else if (type.equals(BluefloodCounterRollup.class))
+            return (AbstractSerializer<T>) counterRollupInstance;
+        else if (type.equals(BluefloodGaugeRollup.class))
+            return (AbstractSerializer<T>)gaugeRollupInstance;
+        else if (type.equals(BluefloodEnumRollup.class))
+            return (AbstractSerializer<T>)enumRollupInstance;
+        else if (type.equals(BluefloodSetRollup.class))
+            return (AbstractSerializer<T>)setRollupInstance;
+        else if (type.equals(SimpleNumber.class))
+            return (AbstractSerializer<T>)fullInstance;
+        else if (type.equals(Integer.class))
+            return (AbstractSerializer<T>)fullInstance;
+        else if (type.equals(Long.class))
+            return (AbstractSerializer<T>)fullInstance;
+        else if (type.equals(Double.class))
+            return (AbstractSerializer<T>)fullInstance;
+        else if (type.equals(Float.class))
+                return (AbstractSerializer<T>)fullInstance;
+        else if (type.equals(byte[].class))
+            return (AbstractSerializer<T>)fullInstance;
+        else if (type.equals(Object.class))
+            return (AbstractSerializer<T>)fullInstance;
+        else
+            return (AbstractSerializer<T>)fullInstance;   
+    }
```

Comment:
In continuation to my comment for AbstractSerDes class, I think all the serializable types(BasicRollUP, SimpleNumber etc) should share a common interface. That would make code look much clean.